File: pizza-store-ui/src/app/core/models/combo.ts
export interface ComboDetail{
  id: number;
  comboId: number;
  productId: number;
  quantity: number;
  productName: string;
  productImageUrl: string;
}

export interface ComboDetailRequest {
  productId: number;
  quantity: number;
}

export interface Combo {
  id: number;
  name: string;
  description: string;
  imageUrl: string;
  price: number;
  isAvailable: boolean;
  isDeleted: boolean;
  createdAt: string;
  details: ComboDetail[];
}

export interface ComboRequest {
  name: string;
  description: string;
  imageUrl: string;
  price: number;
  isAvailable: boolean;
  details: ComboDetailRequest[];
}

File: pizza_store_api/src/main/java/fa/training/pizza_store_api/model/Combo.java
package fa.training.pizza_store_api.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class Combo {
    private int id;
    private String name;
    private String description;
    private String imageUrl;
    private BigDecimal price;
    @JsonProperty("isAvailable")
    private boolean isAvailable;
    @JsonProperty("isDeleted")
    private boolean isDeleted;
    private LocalDateTime createdAt;

    // Danh sách các món ăn thuộc Combo này
    private List<ComboDetail> details;
}
